### ISSUE TITLE
Implement and test adaptive training alogithms

### ISSUE THREAD

**Issue by gorinars** [cmusphinx/g2p-seq2seq#29]
SGD seems to converge slowly. Can we have an option for RMSProp, Adadelta and Adagrad? This should be easy to implement with the respective [Tensorflow optimizers](https://www.tensorflow.org/versions/r0.8/api_docs/python/train.html#RMSPropOptimizer)

**Comment by bmilde** [comment]
I have already tested ADAM on this, convergence is much faster. But the change needs to be made to tensorflows seq2seq model. As you said, its simple: I copied seq2seq_model.py from tensorflow and changed the line with opt into:

opt = tf.train.AdamOptimizer(self.learning_rate)

The learning rate needs to be changed for ADAM, a good default is 0.001

Then you simply change the import in g2p.py:

 #from tensorflow.models.rnn.translate import seq2seq_model
from g2p_seq2seq import seq2seq_model

**Comment by gorinars** [comment]
Folks say RMSProp might be even better for recurrent networks. I did not compare on this particular toolkit. By the way, a pull request is always welcomed ;)
